subworkflows/local/input_check.nf

```diff
@@ -272,3 +278,124 @@ def get_read_counts ( stats ) {
 
     return read_count_meta
 }
+
+/*
+ * Function to validate and resolve BUSCO database paths
+ * Handles the common user error of including '/lineages' at the end of the path
+ */
+def validateBuscoDatabase(db_path) {
+    def path_file = file(db_path)
+    if (path_file.isDirectory()) {
+        // Check if path ends with /lineages and has a parent directory
+        if (path_file.name == 'lineages' && path_file.parent != null) {
+            def parent_dir = file(path_file.parent)
+            log.info "BUSCO path correction: Detected '/lineages' suffix in path"
+            log.info "  Original path: ${path_file}"
+            log.info "  Corrected path: ${parent_dir}"
+            log.info "This prevents the common error where BUSCO tries to use '${path_file}/lineages/lineage_name' instead of '${parent_dir}/lineages/lineage_name'"
+            return parent_dir
+        } else {
+            // Path looks correct, return as-is
+            log.info "Using BUSCO database path: ${path_file}"
+            return path_file
+        }
+    } else {
+        error """
+        ERROR: Invalid BUSCO database path: ${path_file}
+        BUSCO databases must be directories containing the 'lineages/' subdirectory.
+        Please ensure the path points to a valid BUSCO database directory.
+        Common issues:
+        - Path should point to the directory containing 'lineages/' subdirectory
+        - Do NOT include '/lineages' at the end of the path
+        - BUSCO databases cannot be individual files
+        Example: --busco /path/to/busco_downloads/
+        NOT: --busco /path/to/busco_downloads/lineages/
+        """
+    }
+}
+
+/*
+ * Function to validate and resolve BLAST nucleotide database paths
+ * Handles both directory paths (for backwards compatibility) and direct .nal file paths
+ */
+def validateBlastnDatabase(db_path) {
```

Comment:
There's still a problem with the way the function is used. When people mix multiple Blast databases under the same directory (like in #184), yes, they can now refer to one `.nal` specifically, but if only the parent directory makes it to the blastn job, the [module](https://github.com/sanger-tol/blobtoolkit/blob/0.8.0/modules/nf-core/blast/blastn/main.nf#L33) will then still pick up all the `.nal` files.
I think there should be another parameter coming out of the function, to record the name of the `.nal` file, and pass it down to the module.

Reply:
the db_name parameter is now implemented and flows from the validation function to the BLAST module.